includes/class-amp-http.php

```diff
@@ -235,9 +235,15 @@ public static function get_amp_cache_hosts() {
 		 */
 		foreach ( $domains as $domain ) {
 			if ( function_exists( 'idn_to_utf8' ) ) {
-				// The third parameter is set explicitly to prevent issues with newer PHP versions compiled with an old ICU version.
+				$intl_idna_variant = defined( 'INTL_IDNA_VARIANT_UTS46' ) ? INTL_IDNA_VARIANT_UTS46 : false;
+
 				// phpcs:ignore PHPCompatibility.Constants.RemovedConstants.intl_idna_variant_2003Deprecated
-				$domain = idn_to_utf8( $domain, IDNA_DEFAULT, defined( 'INTL_IDNA_VARIANT_UTS46' ) ? INTL_IDNA_VARIANT_UTS46 : INTL_IDNA_VARIANT_2003 );
+				$intl_idna_variant = ( empty( $intl_idna_variant ) && defined( 'INTL_IDNA_VARIANT_2003' ) ) ? INTL_IDNA_VARIANT_2003 : false;
+
+				// The third parameter is set explicitly to prevent issues with newer PHP versions compiled with an old ICU version.
+				if ( ! empty( $intl_idna_variant ) ) {
+					$domain = idn_to_utf8( $domain, IDNA_DEFAULT, $intl_idna_variant );
+				}
```

Comment:
As you noted in chat @dhaval-parekh this change was necessary due to phpstan throwing the error:

```
 ------ --------------------------------------------------------------------
  Line   includes/class-amp-http.php
 ------ --------------------------------------------------------------------
  240    Constant INTL_IDNA_VARIANT_2003 not found.
         💡 Learn more at https://phpstan.org/user-guide/discovering-symbols
 ------ --------------------------------------------------------------------
```

This is not being logged as an error currently as this only occurs on PHP 8. The PHP static analysis GHA currently runs on PHP 7.4 so we'll have to bump that.

Reply:
I've opened ampproject/amp-wap/pull/6466 to bump the PHP version to 8.0, amongst other related changes.

Reply:
Why did this start failing in this PR?

Reply:
If I recall correctly the `INTL_IDNA_VARIANT_2003` constant did not have a `is_defined` check and so phpstan flagged it as it would be  undefined in PHP 8.

Reply:
Oh, and because Dhaval was doing development locally using PHP 8. It wasn't failing on GHA (yet).

Reply:
Indeed.